dash/include/dash/Meta.h

```diff
@@ -90,6 +90,19 @@ struct is_container_compatible :
          >
 { };
 
+/**
+ * Type trait indicating whether a type can be used for global atomic
+ * operations.
+ */
+template <typename T>
+struct is_atomic_compatible
+: public std::integral_constant<
+           bool,
+              dash::is_container_compatible<T>::value
+           && sizeof(T) <= sizeof(std::size_t)
+         >
```

Comment:
That sounds like a horrible idea to me: the user will expect an atomic add on a `double` to behave like an add, not like an add on a pointer type punned to int64_t. This semantic is only useful for `compare_and_swap`, which does not do type-punning at the moment.